File: SPSP/UI/spsp_mobile/lib/models/promotion.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:spsp_mobile/models/menu_item.dart';

part 'promotion.g.dart';

@JsonSerializable()
class Promotion {
  int id;
  DateTime startTime;
  DateTime? endTime;
  String description;
  bool active;
  int? menuItemId;
  MenuItem? menuItem;
  bool? valid;

  Promotion(this.id, this.startTime, this.endTime, this.description, this.active,
      this.menuItemId, this.menuItem, this.valid);

  factory Promotion.fromJson(Map<String, dynamic> json) => _$PromotionFromJson(json);

  Map<String, dynamic> toJson() => _$PromotionToJson(this);
}

File: SPSP/SPSP.Services/Database/Promotion.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace SPSP.Services.Database
{
    public partial class Promotion
    {
        public int Id { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public string? Description { get; set; }
        public bool Active { get; set; }
        public int MenuItemId { get; set; }
        public virtual MenuItem MenuItem { get; set; }
        public bool? Valid { get; set; }

    }
}
